erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -120,6 +120,22 @@ frappe.ui.form.on('Stock Entry', {
 			});
 		}
 
+		if (frm.doc.docstatus == 0 && frm.doc.__onload &&
+			frm.doc.__onload.allow_alternative_item) {
```

Comment:
What if user added an item later after loading the stock entry, which have allows alternative items?

Reply:
Fixed!